### PR TITLE
Support AAD token login in PAI mode

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: docs/en_US/TrainingService/PaiMode.md [hunk 1]
```diff
@@ -82,6 +82,22 @@ Compared with [LocalMode](LocalMode.md) and [RemoteMachineMode](RemoteMachineMod
         portNumber: 1
     ```
 
+NNI support two kind of authorization method in PAI, including password and AAD token (could be viewed from PAI webportal). The authorization is configured in `paiConfig` field.
```
Comment (xuehui1991) [inline docs/en_US/TrainingService/PaiMode.md]: "could be viewed from PAI webportal" --> could you add a link here.

Comment (SparkSnail) [reply docs/en_US/TrainingService/PaiMode.md]: Updated

**Comment by liuzhe-lz** [comment]
Looks much simpler than expected ☺

File: src/nni_manager/training_service/pai/paiTrainingService.ts [hunk 2]
```diff
@@ -256,9 +256,15 @@ class PAITrainingService implements TrainingService {
                     path: '/webhdfs/api/v1',
                     host: this.paiClusterConfig.host
                 });
+                if(this.paiClusterConfig.passWord) {
+                    // Get PAI authentication token
+                    await this.updatePaiToken();
+                } else if(this.paiClusterConfig.token) {
+                    this.paiToken = this.paiClusterConfig.token;
+                } else {
+                    deferred.reject(new Error('pai cluster config format error, please set password or token!'));
```
Comment (chicm-ms) [inline src/nni_manager/training_service/pai/paiTrainingService.ts]: It is unnecessary to use deferred in this method, please help refactor it.

**Pull request merged**